From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Wed, 1 Nov 2023 13:23:35 -0600
Subject: Fix nvidia-ko-to-module-symvers pattern matching

The previously used pattern would hit partial matches, which
could lead to errors if more than one symbol shared the
same prefix.

diff --git a/driver/linux/nvidia-ko-to-module-symvers b/driver/linux/nvidia-ko-to-module-symvers
index c665f9a9..fbfe4961 100755
--- a/driver/linux/nvidia-ko-to-module-symvers
+++ b/driver/linux/nvidia-ko-to-module-symvers
@@ -16,7 +16,7 @@ symvers_fn="$2"
 
 touch "${symvers_fn}"
 for sym in "${syms[@]}"; do
-	crc="$(objdump -t "${nvidia_ko_fn}" | grep "__crc_${sym}" | awk '{print $1}')"
+	crc="$(objdump -t "${nvidia_ko_fn}" | grep "__crc_${sym}$" | awk '{print $1}')"
 	if [ -z "${crc}" ]; then
 		echo "Warning: Can't find symbol ${sym} in ${nvidia_ko_fn}; assuming CONFIG_MODVERSIONS=n so setting CRC=0"
 		crc=00000000
-- 
2.34.1

